business_requirement_sale/wizard/business_requirement_create_sale.py

```diff
@@ -36,17 +52,46 @@ def default_get(self, fields):
         res = super().default_get(fields)
         context = self.env.context
         if ('business_requirement_id' in fields and
-                not res.get('business_requirement_id') and
                 context.get('active_model') == 'business.requirement' and
                 context.get('active_id')):
             res['business_requirement_id'] = context['active_id']
             br = self.env['business.requirement'].browse(context['active_id'])
-            if ('deliverable_ids' in fields and
-                    not res.get('deliverable_ids') and
-                    res.get('business_requirement_id')):
-                res['deliverable_ids'] = [(6, 0, br.deliverable_lines.ids)]
+            if 'deliverable_ids' in fields:
+                if not br.deliverable_lines:
+                    raise exceptions.UserError(_(
+                        'No deliverables found for this business requirement.'
+                    ))
+            if 'has_undefined_section' in fields:
+                res['has_undefined_section'] = any(
+                    not x.section_id for x in br.deliverable_lines
+                )
+            if 'section_ids' in fields:
+                sections = br.mapped('deliverable_lines.section_id')
+                res['applicable_section_ids'] = [(6, 0, sections.ids)]
         return res
 
+    @api.onchange('section_ids')
+    def _onchange_section_ids(self):
+        self.ensure_one()
+        br = self.business_requirement_id
+        new_deliverable_ids = []
+        for section in self.applicable_section_ids:
+            deliverables = br.deliverable_lines.filtered(
+                lambda x: x.section_id == section
+            )
+            if section in self.section_ids:
+                new_deliverable_ids += deliverables.ids
+        self.deliverable_ids = [(6, 0, new_deliverable_ids)]
+
+    @api.onchange('undefined_section')
+    def _onchange_undefined_section(self):
+        self.ensure_one()
+        deliverables = self.business_requirement_id.deliverable_lines.filtered(
+            lambda x: not x.section_id
+        )
+        command = 4 if self.undefined_section else 3
```

Comment:
I would add a comment, explains that deliverables will be added if undefined_section is True or removed if False
```suggestion
        # If undefined_section deliberables with no section will be added
        # otherwise need to be removed
        command = 4 if self.undefined_section else 3
```

Reply:
But this is explaining how m2m commands work, which is not the scope of this code...

Reply:
I just feel better explaining what it does the function if the function name is not declarative. If could be as a docstring would be fine. Nevertheless is a nice to have